### ISSUE TITLE
woocommerce_api_authentication_error","message":"Consumer Secret is invalid

### ISSUE THREAD

**Issue by chbichsel** [woothemes/wc-api-python#16]
When using HTTPS in the API URL parameter below the authentication fails with the error" 
> "woocommerce_api_authentication_error","message":"Consumer Secret is invalid". 

HTTP works fine. 

Questions:
 

- Any suggestions on how to troubleshoot this?

- If this is a potential server side configuration issues, do you have any pointers on how to configure it correctly?

- Could you give me an example of how to use query parameters instead? The [WooCommerce REST API documentation](http://woothemes.github.io/woocommerce-rest-api-docs/#authentication-over-https) hints at an example for servers that not properly parse the Authorization header, but it is actually missing. 

```
API(
            url="https://kashalife.net/",
            consumer_key   ="ck_16c59a1a0693c2723928830da52a30640c2311fd",
            consumer_secret="cs_0b4e0f17741e024b31d0ef799257b972db473305",
            wp_api=False,
            verify_ssl=False,
            version="v3",
        )
```

Thanks - Christoph

**Comment by claudiosmweb** [comment]
Maybe it's your server's configuration. Have you saw our docs?

> Occasionally some servers may not parse the Authorization header correctly (if you see a “Consumer key is missing” error when authenticating over SSL, you have a server issue). In this case, you may provide the consumer key/secret as query string parameters instead.

You can test using cURL:

```bash
curl -k -u consumer_key:consumer_secret https://www.example.com/wp-json/wc/v1/orders
```

And 

```bash
curl -k https://www.example.com/wp-json/wc/v1/orders?consumer_key=123&consumer_secret=abc
```

**Comment by chbichsel** [comment]
Hi Claudio, 

Thanks for your reply! Before posting this questions I have been trying for 5+ days to figure this out by myself by using documentations from woothemes, woocommerce, and any other info on google. The fact that I have not been able to resolve the issue, and have not found a lot of detail on it, tells me that I am either searching for the wrong things , or that it is so obvious for people who understand basic auth over https and webserver configurations that its just a no brainer (for them). 

Two questions:

1. Thanks for the curl commands. Auth with _ULR params works with my server, but not headers_. I haven't tried "Websites behind load balancers or reverse proxies" of the [WooThemes SSL and HTTPS documentation](https://docs.woothemes.com/document/ssl-and-https/). I'll do that next. Any other pointers to what configuration may be wrong?

2. URL params auth is not natively supported in the python wrapper, correct? 

![image](https://cloud.githubusercontent.com/assets/18748109/16276336/bedfa34a-3862-11e6-8e98-95a9d33b23c0.png)


```
** Header auth wc-API (fails)
url -k -u ck_6beecb5aad00b590d8e69d23ccf59773d002f6e5:cs_e6ba5dbce62af9c0f92ebbaa78557177135688de https://kashalife.net/wc-api/v3/products/

* Header auth wp-JSON (fails)
curl -k -u ck_6beecb5aad00b590d8e69d23ccf59773d002f6e5:cs_e6ba5dbce62af9c0f92ebbaa78557177135688de https://kashalife.net/wp-json/wc/v1/products/

URL PARAMS wc-API (ok)
curl -k -g "https://kashalife.net/wc-api/v3/products/?consumer_key=ck_6beecb5aad00b590d8e69d23ccf59773d002f6e5&consumer_secret=cs_e6ba5dbce62af9c0f92ebbaa78557177135688de"

URL PARAMS wc-JSON (ok)
curl -k -g "https://kashalife.net/wp-json/wc/v1/products/?consumer_key=ck_6beecb5aad00b590d8e69d23ccf59773d002f6e5&consumer_secret=cs_e6ba5dbce62af9c0f92ebbaa78557177135688de"
```

**Comment by claudiosmweb** [comment]
> URL params auth is not natively supported in the python wrapper, correct? 

No yet, I added support for it in our other wrappers.
I'll add support for it today, by the end of the day you'll be able to update and use it as query strings :)

**Comment by chbichsel** [comment]
Wow, great! Thank you Claudio!

**Comment by claudiosmweb** [comment]
@chbichsel Solved in version 1.2.0, now you just need to update and set your wrapper using the `query_string_auth` option, see example:

```python
API(
    url="https://kashalife.net/",
    consumer_key="ck_16c59a1a0693c2723928830da52a30640c2311fd",
    consumer_secret="cs_0b4e0f17741e024b31d0ef799257b972db473305",
    wp_api=False,
    verify_ssl=False,
    version="v3",
    query_string_auth=True
)
```

**Comment by chbichsel** [comment]
Sweet. Thanks!
